oaknational/Oak-Web-Application
# Feat/lesq 411/loading button [LESQ-411]

## Description

- Add a new button `LoadingButton` using new button design from figma
- Add a new button border component `DoubleButtonBorders` which adds an additional double outline on focus
- Replace button on the download with the new LoadingButton
- Disable button on downloads page when there are form errors or an incomplete form

## How to test

1. Go to https://deploy-preview-2058--oak-web-application.netlify.thenational.academy/teachers/programmes/combined-science-secondary-ks4-foundation-ocr/units/measuring-waves/lessons/transverse-waves/downloads?preselected=all
2. Click on Download
3. You should see the button update with a loading state and revert on success
5. Also check the component  on storybook and toggle between loading/enter different text - it's in the Button section and called `Loading`

## Screenshots

How it used to look (delete if n/a):
<img width="303" alt="Screenshot 2023-10-30 at 13 14 34" src="https://github.com/oaknational/Oak-Web-Application/assets/45038878/4ed9082e-a206-41fc-80eb-7478442c3532">

How it should now look:
<img width="303" alt="Screenshot 2023-10-30 at 13 15 16" src="https://github.com/oaknational/Oak-Web-Application/assets/45038878/f3c399cc-1787-42b2-82b7-205dbad2d3f6">
<img width="303" alt="Screenshot 2023-10-30 at 15 13 09" src="https://github.com/oaknational/Oak-Web-Application/assets/45038878/061b429c-e8fd-4477-a860-b4be6187a06f">
<img width="303" alt="Screenshot 2023-10-30 at 15 17 35" src="https://github.com/oaknational/Oak-Web-Application/assets/45038878/23d870ca-f7d2-4f4f-afd1-4d170db041d9">

## ACs
- [ ]  Use disabled button style (in grey) with an animated spinner (animation like the one in the second row, in the middle here: https://loading.io/css/)
- [ ]  Accessibility: if user has motion reduction turned on, ensure the animation doesn’t work
- [ ]  Button copy to say ‘`Downloading…`’
- [ ]  Button has a `disabled` attribute on it, so it can’t be clicked while in the loading state, and this is communicated as such to screen readers and other assistive tech
- [ ]  Button is re-enabled after a successful (or failed) download, with its original button text
